File: Python/random_or_unmarked/simple_calculator.py
def get_float_input(prompt):
    while True:
        try:
            value = float(input(prompt))
            return value
        except ValueError as ve:
            print(f"Value Error: {ve}")


def get_operator(prompt):
    while True:
        operator = input(prompt)
        if operator in ["+", "-", "*", "/"]:
            return operator
        print("Invalid operator. Enter a valid operator.")


def calculate_result(num1, num2, operator):
    if operator == "+":
        return num1 + num2
    elif operator == "-":
        return num1 - num2
    elif operator == "*":
        return num1 * num2
    else:
        return num1 / num2


def main():
    while True:
        num1 = get_float_input("Enter first number: ")
        num2 = get_float_input("Enter second number: ")
        operator = get_operator("Enter operator (+, -, *, /): ")
        try:
            result = calculate_result(num1, num2, operator)
            print(f"Result: {result}")
            break
        except ZeroDivisionError as zde:
            print(f"Zero Division Error: {zde}")
        except Exception as e:
            print(f"An error occurred: {e}")


if __name__ == "__main__":
    main()


"""""
Old code, ignore

while True:
    try:
        num1 = float(input("Enter first number: "))
        num2 = float(input("Enter second number: "))
        operation = input("Enter operator (+, -, *, /): ")

        if operation not in ["+", "-", "*", "/"]:
            raise ValueError("Invalid operator.")
        
        if operation == "+":
            result = num1 + num2
        elif operation == "-":
            result = num1 - num2
        elif operation == "*":
            result = num1 * num2
        else:
            result = num1 / num2

        print(f"Result: {result}")
        break

    except ValueError as ve:
        print(f"Value Error: {ve}")
    except ZeroDivisionError as zde:
        print(f"Zero Division Error: {zde}")
    except Exception as e:
        print(f"An error occured: {e}")
"""""

File: C/random_or_unmarked/simpleCalculator.c
#include <stdio.h>
#include <stdbool.h>
#include <ctype.h>
#include <stdlib.h>
#include <string.h>

void clear_input_buffer() {
    int c;
    while ((c = getchar()) != '\n' && c != EOF);
}

double get_valid_number(const char *prompt) {
    double num;
    char input[100];
    while (true) {
        printf("%s", prompt);
        if (fgets(input, sizeof(input), stdin) != NULL) {
            if (sscanf(input, "%lf", &num) == 1) {
                return num;
            } else {
                printf("Invalid input. Please enter a valid number.\n");
            }
        } else {
            printf("Error reading input. Please try again.\n");
        }
    }
}

char get_valid_operator(const char *prompt) {
    char op;

    while (true) {
        printf("%s", prompt);
        op = getchar();
        clear_input_buffer();
        if (op == '+' || op == '-' || op == '*' || op == '/') {
            return op;
        } else {
            printf("Invalid operator. Please enter one of (+, -, *, /).\n");
        }
    }
}

int main() {
    double num1, num2;
    char operation;

    while (true) {
        num1 = get_valid_number("Enter first number: ");
        num2 = get_valid_number("Enter second number: ");
        operation = get_valid_operator("Enter operator (+, -, *, /): ");

        switch (operation) {
            case '+':
                printf("Result: %lf\n", num1 + num2);
                break;
            case '-':
                printf("Result: %lf\n", num1 - num2);
                break;
            case '*':
                printf("Result: %lf\n", num1 * num2);
                break;
            case '/':
                if (num2 != 0) {
                    printf("Result: %lf\n", num1 / num2);
                } else {
                    printf("Error: Division by zero is not allowed.\n");
                }
                break;
        }

        char choice;
        printf("Do you want to perform another calculation? (Y/N): ");
        choice = getchar();
        clear_input_buffer();
        if (toupper(choice) != 'Y') {
            printf("Goodbye!\n");
            break;
        }
    }

    return 0;
}
